---
title: "scRNA-seq"
subtitle: "UMAP/Clustering"
author: "Sung Rye Park"
date: "`r format(Sys.Date())`"
output:  
  rmdformats::robobook: 
    code_folding: hide 
    number_sections: FALSE
    toc_depth: 6
    toc_float: false
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=F, fig.align = "left", 
                      message=F, warning=F,
                      results = "markup",
                      error = TRUE,
                      highlight = TRUE,
                      prompt = FALSE,
                      tidy = FALSE)
```

```{r}
## load required packages
library(Seurat)
library(cowplot)
library(dplyr)
library(ggplot2)
library(DT)
library(paletteer)
library(viridis)
```

<br>

Patient data    

* Filtering  

    * UMI >= 1000  
    * Mitochondrial content <= 10  
    * Doublet removal : Yes (scrublet)   

```{r, eval=FALSE}
dir= "~/Desktop/DF/DFCI_Paweletz/2024_Daiichi_DXD/"
obj.srt = readRDS(paste0(dir,"rds/P30342.init.24.05.06.rds"))
```


```{r, eval=FALSE}
obj.srt = subset(obj.srt, nCount_RNA >= 1000 & percent.mt <= 10 & predicted_doublet == "False")
```



```{r, eval=FALSE}
## perform default analysis
perform_default_analysis <- function(obj.srt, n_features = 2000, n_pcs = 20, 
                                     dims_for_neighbors = 1:20, 
                                     resolutions = c(0.2, 0.4), 
                                     umap_dims = 1:20) {
  # Step 1: Find variable features
  obj.srt <- FindVariableFeatures(obj.srt, 
                                  selection.method = 'vst', 
                                  nfeatures = n_features)
  
  # Step 2: Scale and normalize data
  all_genes <- rownames(obj.srt)
  obj.srt <- NormalizeData(obj.srt)
  obj.srt <- ScaleData(obj.srt, features = all_genes)
  
  # Step 3: Run PCA
  obj.srt <- RunPCA(obj.srt, 
                    features = VariableFeatures(object = obj.srt), npcs = n_pcs)
  
  # Step 4: Find neighbors
  obj.srt <- FindNeighbors(obj.srt, dims = dims_for_neighbors)
  
  # Step 5: Find clusters
  obj.srt <- FindClusters(obj.srt, resolution = resolutions)
  
  # Step 6: Run UMAP
  obj.srt <- RunUMAP(obj.srt, dims = umap_dims)
  
  # Return the Seurat object with analysis results
  return(obj.srt)
}

# apply
obj.srt <- perform_default_analysis(obj.srt)
```



```{r, eval=FALSE}
obj.srt %>% saveRDS(paste0(dir,"rds/P30342.24.05.07.rds"))
```


```{r}
dir= "~/Desktop/DF/DFCI_Paweletz/2024_Daiichi_DXD/"
obj.srt = readRDS(paste0(dir,"rds/P30342.24.05.07.rds"))
```

```{r}
# Modify sample names to hide the project information 
obj.srt@meta.data <- obj.srt@meta.data %>%
  mutate(sample = case_when(
    orig.ident == paste0("P30342_", "M") ~ "CTRL",
    orig.ident == paste0("P30342_", "D") ~ "CND1",
    orig.ident == paste0("P30342_", "I") ~ "CND2",
    orig.ident == paste0("P30342_", "A") ~ "CND3",
    TRUE ~ orig.ident  # 기본값 설정, 일치하지 않는 경우 원래의 값 유지
  ))
obj.srt@meta.data$sample = factor(obj.srt@meta.data$sample, 
                                  levels = c("CTRL", "CND1","CND2","CND3"))
```

# UMAP {.tabset}

## UMAP by treatment

```{r}
sample_cols = scale_color_paletteer_d("ggsci::nrc_npg") 
sample_fils = scale_fill_paletteer_d("ggsci::nrc_npg") 
```


```{r, fig.width=7, fig.height=6, out.width='80%'}
DimPlot(obj.srt, group.by = 'sample', alpha = 0.8) + scale_color_paletteer_d("ggsci::nrc_npg") + theme_bw()
```

## Split by sample  
```{r, fig.width=11, fig.height=9}
DimPlot(obj.srt, group.by = 'sample', split.by = 'sample', ncol = 2, alpha = 0.5) + scale_color_paletteer_d("ggsci::nrc_npg") + theme_bw()
```

## Distribution of number of cells

```{r, fig.width=5, fig.height=4}
# obj.srt@meta.data %>% select(sample) %>% table() %>% data.frame() %>% DT::datatable(caption = 'Number of Cells by sample')
obj.srt@meta.data %>% ggplot(aes(sample, fill=sample)) + 
  geom_bar(alpha=0.7, color="grey5", size=0.1) +
  geom_text(stat="count", aes(label= ..count..), vjust=-0.5, size=3) +
  sample_fils + xlab("") + 
  theme_classic() +
  theme(legend.title = element_blank(),
        axis.text.x = element_text(angle = 45, vjust=0.5)) 

```


# Clustering {.tabset} 

* resolution 0.2  
* resolution 0.4  
<br>

```{r}
resolution_values <- c(0.2, 0.4)
```

## Resolution 0.2 UMAP

```{r}
# Resolution and color palette
i=1
res= paste0("RNA_snn_res.", resolution_values[i])
palette <- wesanderson::wes_palette("FantasticFox1", length(levels(obj.srt@meta.data[,res])), type = "continuous")
```



```{r,fig.width=7, fig.height=6, out.width='80%'}
DimPlot(obj.srt, group.by = res, cols = palette, alpha = 0.8) + 
  theme(plot.title = element_blank()) + xlab("UMAP1") + ylab("UMAP2")
```

## Resolution 0.2 UMAP (label)
```{r,fig.width=7, fig.height=6, out.width='80%'}
DimPlot(obj.srt, group.by = res, alpha = 0.8,
        label = T, label.box = T, label.size = 3, cols = palette) + theme(plot.title = element_blank())
```


## Cell numbers 
```{r}

obj.srt@meta.data %>% ggplot(aes(!!sym(res), fill=!!sym(res))) + 
  geom_bar(alpha=0.7, color="grey5", size=0.1) +
  geom_text(stat="count", aes(label= ..count..), vjust=-0.5, size=3) +
  scale_fill_manual(values=palette) + 
  xlab("") + 
  theme_classic() +
  theme(legend.title = element_blank(),
        axis.text.x = element_text(angle = 45, vjust=0.5)) +
  ggtitle("Cluster")


obj.srt@meta.data %>% select(!!sym(res)) %>% table() %>% data.frame() %>%
  DT::datatable(caption = 'Number of Cells by clusters',
    options = list(pageLength = length(levels(obj.srt@meta.data[[res]]))), rownames = F)
obj.srt@meta.data %>%
  select(!!sym(res), sample) %>%
  table() %>%
  data.frame() %>%
  tidyr::spread(sample, Freq) %>%
  DT::datatable(
    caption = 'Number of Cells by clusters and treatment',
    options = list(pageLength = length(levels(obj.srt@meta.data[[res]]))), rownames = F
  )
```


## Proportion by sample
```{r, fig.width=8, fig.height=4}
obj.srt@meta.data %>% ggplot(aes(sample, fill=!!sym(res))) +
  geom_bar(position = "fill", color="grey9",size = 0.2) +coord_flip() +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=palette) +xlab("") +ylab("Fraction")
```

## Proportion by cluster 
```{r}
obj.srt@meta.data %>% ggplot(aes(!!sym(res), fill=sample)) + 
  geom_bar(position = "fill", color="grey9",size = 0.2, alpha=0.8) + scale_fill_paletteer_d("ggsci::nrc_npg") + theme_classic()+ 
  theme(legend.title = element_blank()) +
  ylab("Fraction")

```





## Resolution 0.4 UMAP

```{r}
# Resolution and color palette
i=2
res= paste0("RNA_snn_res.", resolution_values[i])
palette <- wesanderson::wes_palette("FantasticFox1", length(levels(obj.srt@meta.data[,res])), type = "continuous")
```


```{r,fig.width=7, fig.height=6, out.width='80%'}
DimPlot(obj.srt, group.by = res, cols = palette, alpha = 0.8) + 
  theme(plot.title = element_blank()) + xlab("UMAP1") + ylab("UMAP2")
```

## Resolution 0.4 UMAP (label)
```{r,fig.width=7, fig.height=6, out.width='80%'}
DimPlot(obj.srt, group.by = res, alpha = 0.8,
        label = T, label.box = T, label.size = 3, cols = palette) + theme(plot.title = element_blank())
```

## Cell numbers 
```{r}

obj.srt@meta.data %>% ggplot(aes(!!sym(res), fill=!!sym(res))) + 
  geom_bar(alpha=0.7, color="grey5", size=0.1) +
  geom_text(stat="count", aes(label= ..count..), vjust=-0.5, size=3) +
  scale_fill_manual(values=palette) + 
  xlab("") + 
  theme_classic() +
  theme(legend.title = element_blank(),
        axis.text.x = element_text(angle = 45, vjust=0.5)) +
  ggtitle("Cluster")
```


## Proportion by sample
```{r, fig.width=8, fig.height=4}
obj.srt@meta.data %>% ggplot(aes(sample, fill=!!sym(res))) +
  geom_bar(position = "fill", color="grey9",size = 0.2) +coord_flip() +
  scale_fill_manual(values=palette) +xlab("") +ylab("Fraction")
```

## Proportion by cluster 
```{r}
obj.srt@meta.data %>% ggplot(aes(!!sym(res), fill=sample)) + 
  geom_bar(position = "fill", color="grey9",size = 0.2, alpha=0.8) + scale_fill_paletteer_d("ggsci::nrc_npg") + theme_classic()+ 
  theme(legend.title = element_blank()) +
  ylab("Fraction")

```

## Sankey plot 

```{r}
## sankey plot 
library(ggsankey)
library(ggplot2)
library(dplyr)

cols = c('sample','RNA_snn_res.0.2','RNA_snn_res.0.4')
data =obj.srt@meta.data[, cols]

# Function to create a Sankey plot
create_sankey_plot <- function(data, column_names, title, show_labels = FALSE){
  df <- data %>%
    make_long(!!column_names[1], !!column_names[2], !!column_names[3])
  
  dagg <- df %>%
    group_by(node) %>%
    tally()
  
  df2 <- merge(df, dagg, by.x = 'node', by.y = 'node', all.x = TRUE)
  
  pl <- ggplot(df2, aes(x = x,
                        next_x = next_x,
                        node = node,
                        next_node = next_node,
                        fill = factor(node),
                        label = paste0(node," n=", n))
  )
  
  pl <- pl + geom_sankey(flow.alpha = 0.5, color = "gray40", show.legend = show_labels)
  if (show_labels) {
    pl <- pl + geom_sankey_label(size = 3, color = "white", fill = "gray40", hjust = 1)  # Change hjust value to 1 (right-aligned)
  }
  
  pl <- pl + theme_bw()
  pl <- pl + theme(legend.position = "none")
  pl <- pl + theme(axis.title = element_blank(),
                   axis.text.y = element_blank(),
                   axis.ticks = element_blank(),
                   panel.grid = element_blank())
  pl <- pl + scale_fill_viridis_d(option = "plasma")
  pl <- pl + labs(title = "")
  
  pl <- pl + labs(fill = 'Nodes')
  
  return(pl)
}

# Assuming obj.srt@meta.data is your data and you want to use specific columns for the Sankey plot
column_names <- cols

# Create the Sankey plot using the specified columns
sankey_plot <- create_sankey_plot(data = obj.srt@meta.data, column_names, 
                                  title = "Test", show_labels = TRUE)
```


```{r,fig.height=6, fig.width=10}
# Display the plot
print(sankey_plot)
```


# Markers {.tabset}

* resolution 0.2  
* resolution 0.4  
<br>

```{r}
resolution_values <- c(0.2, 0.4)
```

```{r, eval=FALSE, echo=F}
# Marker gene Identification
find_and_save_markers <- function(obj.srt, cluster_id, logfc_threshold = 1.2,
                                  test_method = 'wilcox', min_percent = 0.25) {
  Idents(obj.srt) = cluster_id
  all.markers = FindAllMarkers(obj.srt, logfc.threshold = log2(logfc_threshold),
                               only.pos = TRUE,
                               test.use = test_method, min.pct = min_percent)
  return(all.markers)
}
# apply:

resolution_values <- c(0.2, 0.4)

# Loop through each resolution value
for (resolution_number in resolution_values) {
  # Construct the resolution name
  res <- paste0("RNA_snn_res.", resolution_number)
  all.markers= find_and_save_markers(obj.srt= obj.srt, cluster_id = res,
                                   logfc_threshold = 1.2, test_method = 'wilcox', min_percent = 0.25)
  output_file = paste0(dir,"data/mks/30342/Daiichi_30342_", res, ".markers.csv")
  write.csv(all.markers, file = output_file, row.names = T)
}
```


## Markers (0.2) 

**Download cluster marker file**  
(no download on example version)  
```{r}
resolution_number = 0.2
res= paste0("RNA_snn_res.", resolution_number)
mks = read.csv(paste0(dir,"data/mks/30342/Daiichi_30342_", res, ".markers.csv"), row.names = 1)
len =length(levels(obj.srt@meta.data[,res]))
DT::datatable(mks, editable = TRUE,  options = list(dom="Bfrtip", buttons=c("csv","excel"), pageLength=len))
mks.res02 = mks
```

## Heatmap (res 0.2) 

Top 5 genes from each cluster   



```{r, fig.width=14, fig.height=8}
mks = mks.res02
genes = mks %>% group_by(cluster) %>% top_n(5, avg_log2FC) %>% select(gene) %>% pull()
Idents(obj.srt) = res

i=1
res= paste0("RNA_snn_res.", resolution_values[i])
palette <- wesanderson::wes_palette("FantasticFox1", length(levels(obj.srt@meta.data[,res])), type = "continuous")


DoHeatmap(obj.srt, features = genes, group.colors = palette) +
  scale_fill_viridis(option="magma") + 
  theme(axis.title.x.top = element_text(size = 4))
```



## Markers (0.4) 

**Download cluster marker file**  
(no download on example version)  
```{r}
resolution_number = 0.4
res= paste0("RNA_snn_res.", resolution_number)
mks = read.csv(paste0(dir,"data/mks/30342/Daiichi_30342_", res, ".markers.csv"), row.names = 1)
len =length(levels(obj.srt@meta.data[,res]))
DT::datatable(mks, editable = TRUE,  options = list(dom="Bfrtip", buttons=c("csv","excel"), pageLength=len))
mks.res04 = mks
```

## Heatmap (res 0.4) 

Top 5 genes from each cluster   



```{r, fig.width=14, fig.height=10}
mks = mks.res04
genes = mks %>% group_by(cluster) %>% top_n(5, avg_log2FC) %>% select(gene) %>% pull()
Idents(obj.srt) = res

i=2
res= paste0("RNA_snn_res.", resolution_values[i])
palette <- wesanderson::wes_palette("FantasticFox1", length(levels(obj.srt@meta.data[,res])), type = "continuous")


DoHeatmap(obj.srt, features = genes, group.colors = palette) +
  scale_fill_viridis(option="magma") + 
  theme(axis.title.x.top = element_text(size = 4))
```


